blob/blob.go

```diff
@@ -299,8 +303,10 @@ func (o *ListObject) As(i interface{}) bool {
 	return o.asFunc(i)
 }
 
-// Bucket manages the underlying blob service and provides read, write and delete
-// operations on objects within it.
+// Bucket provides an easy and portable way to interact with blobs
+// within a "bucket", including read, write, and list operations.
+// To create a Bucket, use constructors found in provider-specific
```

Comment:
Do we define "provider" anywhere? Or define or label "provider-specific subpacke[s]" ? I'm not sure this term is completely obvious with respect to where I would want to go next if I were first reading it. 

I think an 'eg' and an example of calling one, or a label at the top of the provider-specific packages that says "this is the <SERVICE> provider-specific package", or a definition in DESIGN.md, or some combination thereof would make it clearer.

Reply:
I tried to introduce this better here, PTAL. I think with the example most people will understand what "provider" means.